docs/specifications/tests/Connectivity-TP/connectivity03.md

```diff
@@ -1,58 +1,204 @@
-all of them. 
+The objective in this test is to verify that all IP addresses of the domain's
+authoritative name servers are not announced from the same ASN (autonomous system 
+number). See [RFC 1930] and [Wikipedia] for an explanation of AS (autonomous 
+system).
 
-The objective in this test is to check that all IP addresses of the domain's
-authoritative name servers does not belong to the same AS.
+This test is done separately on IPv4 and IPv6, and both must match the criterion.
 
-[missing references to AS numbers, RFC 1930?]
+[RFC 2182], section 3.1 clearly specifies that distinct authoritative name 
+servers for a child domain should be placed in different topological and 
+geographical locations. The objective is to minimise the likelihood of a single 
+failure disabling all of them. 
 
-### Inputs
 
-The domain name to be tested.
+## Inputs
 
-### Ordered description of steps to be taken to execute the test case
+* "Child Zone" - The domain name to be tested.
+* "ASN Database" - The database of ASN data to be used. Possible values
+  are "RIPE" and "Cymru" (the default value).
+* "Cymru Base Name" - If the *ASN Database* is "Cymru", the default value 
+  is "asnlookup.zonemaster.net".
+* "Ris Whois Server" - If the *ASN Database* is "RIPE", the default value 
+  is "riswhois.ripe.net".
 
-1. Obtain the IP addresses of the name servers from [Method4](../Methods.md).
-2. Obtain the list of ASN lookup domains from the configuration of Zonemaster.
-3. For obtaining the AS numbers for IPv6 addresses; first reverse its eight
-4-character hexadecimal numbers and place dots between each. It is important 
-all omitted zeroes are included.  
-3.1. At the end of the reversed IPv6 address, concatenate the string (from
-step 2)  
-3.2 Send a "TXT" query using the string (obtained from Step 3.1)  
-3.3 If there is an ANSWER, then go to step 3.4, else go to the next string
-in the list (from step 2)  
-3.4 The AS number for the IPv6 address is found in the ANSWER for the query 
-4. To obtain the AS numbers for IPv4 addresses; first reverse the four numerical
-octets and place dots between each.  
-4.1. At the end of the reversed IPv4 address concatenate the string (from step 2)  
-4.2 Send a "TXT" query using the string (obtained from Step 4.1)  
-4.3 If there is an ANSWER, go to step 4.4, else go to the next string
-in the list (from step 2)  
-4.4 The AS number for the IPv4 address is found in the ANSWER for the query 
-5. If all the retrieved AS (obtained from step 3.3 and 4.3) are same, then the test
-   fails.
 
-### Outcome(s)
+## Ordered description of steps to be taken to execute the test case
 
-If there is a AS which is different from the other retrieved AS, then the
-test succeeds.
+1. Obtain the total set of IP addresses of the name servers for the 
+   *Child Zone* using [Method4] and [Method5] and do:
+   1. Create a possibly empty set of the IPv4 addresses ("NS IPv4").
```

Comment:
Not clear why do you want to create an empty set of IP addresses

Reply:
It is more like saying that there might not be any IPv4 addresses. And if the set is not kind of created (but empty) then some could complain about the logic in the step below where there is a reference to the set "NS IPv4".

Do you think it makes it harder to follow or understand the steps?

Reply:
Have you considered something like this?

> Split the total set into two subsets, one for IPv4 addresses ("NS IPv4") and one for IPv6 addresses ("NS IPv6"). (The subsets may be empty.)